### PR TITLE
Finish port to Win32 and polish build system

### PR DESCRIPTION
**Pull request by Botje**
This PR fixes nearly all warnings that occur in a Win32 build.
This includes warnings about file handles on Win32 being `SOCKET` instead of `int`.

Two patches fix generic bugs:
- 1ae7e2e Fixes an issue where TXT records over 63 bytes would get truncated to NULL and crash.
- f684d7c Fixes a `=` in an assert.

### CODE REVIEW

File: mdns.c [hunk 1]
```diff
@@ -146,8 +146,6 @@ uint8_t *create_label(const char *txt) {
 
 	assert(txt != NULL);
 	len = strlen(txt);
-	if (len > 63)
```
Comment (Pro) [inline mdns.c:149]: Why did you remove this check?

**Review by Pro — CHANGES_REQUESTED** [review]

**Comment by Pro** [comment]
Please also merge current master into this PR, I just fixed the missing package.
Then make sure Travis and AppVeyor builds run through without any error.

File: mdns.c (hunk 1 above)
Comment (Botje) [reply mdns.c:149]: If you look at `rr_add_txt` in `mdns.c`, `create_label` is also used to fill in the child entries of a TXT record. Without this change, TXT entries longer than 63 chars are truncated to NULL and the server crashes when it attempts to encode such entries in a packet.

Maybe it is better to revert this and add a `create_txt_label` that does not have this check?

**Comment by Botje** [comment]
I fixed all the compiler errors and warnings I could find myself and then one more found by Travis.
I backed out the change in `create_label` and made a new `create_txt_label` in response to your request.

If you are okay with merging this, I will rebase this branch to get rid of the reverted commit and fixup commits and force push it over this branch.

**Review by Pro — APPROVED** [review]

**Comment by Pro** [comment]
Thanks, nice work! Looks good to me.

You can also just squash all the commits into one. Please also make sure that your user.email is set to the one used on github: https://help.github.com/articles/setting-your-email-in-git/
In your commits these are different.
